### PR TITLE
TOPAZ: Problem with peak intensity assignment for satellite peaks

### PR DESCRIPTION
**Pull request by jmborr**
**Work done**:  
- [x] New event search around peaks in the QLab frame. Doesn't differentiate between main and satellite peaks, implemented in `integrateQLabEvents`.
- [x] `IntegrateEllipsoids` modifed to use `integrateQLabEvens`.
- [x] Unit test for `integrateQLabEvents` and system test for `IntegrateEllipsoids`.

**To test:**
Run the system test downloading the following input files in the `/tmp` directory:
- [TOPAZ_39037_bank29.nxs](https://www.dropbox.com/s/5s6qlpgrci2fj0o/TOPAZ_39037_bank29.nxs?dl=0)
- [TOPAZ_39037_peaks_short.nxs](https://www.dropbox.com/s/kalr5hudyo0wvpp/TOPAZ_39037_peaks_short.nxs?dl=0)
```python
from mantid.simpleapi import *
LoadNexus(Filename='/tmp/TOPAZ_39037_bank29.nxs', OutputWorkspace='events')
LoadNexus(Filename='/tmp/TOPAZ_39037_peaks_short.nxs', OutputWorkspace='peaks_input')
IntegrateEllipsoids(InputWorkspace='events',
                    PeaksWorkspace='peaks_input',
                    OutputWorkspace='peaks_output',
                    RegionRadius=0.14,
                    SpecifySize=True,
                    PeakSize=0.07,
                    BackgroundInnerSize=0.09,
                    BackgroundOuterSize=0.11,
                    CutoffIsigI=5.0,
                    AdaptiveQBackground=True,
                    AdaptiveQMultiplier=0.001,
                    UseOnePercentBackgroundCorrection=False)
```
Verify that the two entries in column `intens` are `938` and `13936` when rounded to the nearest integer.

Refs #30737
---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by KedoKudo — DISMISSED** [review]
The provided testing data and script worked are tested and passed on analysis cluster (RHEL7). I do have a few questions (see comments below), but none of those is critical enough to block this PR.

File: Framework/MDAlgorithms/inc/MantidMDAlgorithms/IntegrateQLabEvents.h
Comment (jmborr) [reply Framework/MDAlgorithms/inc/MantidMDAlgorithms/IntegrateQLabEvents.h:86]: Some code is new, some code is a duplication of code `Integrate3DEllipsoids.cpp`. I left `Integrate3DEllipsoids.cpp` intact because it's being used by algorithm `IntegrateEllipsoidsTwoSteps`

File: Testing/SystemTests/tests/framework/IntegrateEllipsoidsTest.py
Comment (jmborr) [reply Testing/SystemTests/tests/framework/IntegrateEllipsoidsTest.py:21]: Ouch! this was only for debugging purposes and I forgot to remove it  :face_with_head_bandage: 
Good catch!

File: Framework/MDAlgorithms/test/IntegrateEllipsoidsTest.h
Comment (jmborr) [reply Framework/MDAlgorithms/test/IntegrateEllipsoidsTest.h:264]: It's actually not necessary. The old `IntegrateEllipsoids` could integrate also in HKL space and UB inverse matrix was used to convert Q-vectors from QLab space to HKL space. The current `IntegrateEllipsoids` integrates in QLab space.

File: Framework/MDAlgorithms/src/IntegrateEllipsoids.cpp
Comment (jmborr) [reply Framework/MDAlgorithms/src/IntegrateEllipsoids.cpp:320]: This algorithm is used only in ORNL and we already had a conversation with the instrument scientists. They decided they wanted a rewrite of the algorithm instead of a new version. Plus I got the impression that the satellite options were more a hindrance than an advantage to them :smile:

File: Framework/MDAlgorithms/src/IntegrateQLabEvents.cpp
Comment (jmborr) [reply Framework/MDAlgorithms/src/IntegrateQLabEvents.cpp:274]: This is one of them pieces of code I duplicated from `Integrate3DEvents.cpp`. I didn't' look into the cause for the coefficient

**Review by KedoKudo — APPROVED** [review]
Approve again after a minor change.

File: Framework/MDAlgorithms/src/IntegrateQLabEvents.cpp
Comment (peterfpeterson) [reply Framework/MDAlgorithms/src/IntegrateQLabEvents.cpp:274]: As an FYI: you can `constexpr` the factor as a variable in an anonymous namespace so you can document what it is/does and show the derivation of it much more clearly.

**Pull request merged**
